test/src/unit-constructor1.cpp

```diff
@@ -239,6 +239,31 @@ TEST_CASE("constructors")
             CHECK(j == j_reference);
         }
 
+        SECTION("std::pair")
+        {
+            std::pair<float, std::string> p{1.0, "string"};
+            json j(p);
+
+            CHECK(j.type() == json::value_t::array);
+            REQUIRE(j.size() == 2);
+            CHECK(j[0] == std::get<0>(p));
+            CHECK(j[1] == std::get<1>(p));
+        }
+
+        SECTION("std::tuple")
+        {
+            const auto t = std::make_tuple(1.0, "string", 42, std::vector<int> {0, 1});
+            json j(t);
+
+            CHECK(j.type() == json::value_t::array);
+            REQUIRE(j.size() == 4);
+            CHECK(j[0] == std::get<0>(t));
+            CHECK(j[1] == std::get<1>(t));
+            CHECK(j[2] == std::get<2>(t));
+            CHECK(j[3][0] == 0);
+            CHECK(j[3][1] == 1);
+        }
+
```

Comment:
Could you also add a test case for the case when we have an array with, say, 3 elements and ask for a `std::pair` and something similar with tuples?